src/libutil/config.hh

```diff
@@ -232,16 +232,19 @@ protected:
 
     T value;
     const T defaultValue;
+    const bool documentDefault;
 
 public:
 
     BaseSetting(const T & def,
+        const bool documentDefault,
         const std::string & name,
         const std::string & description,
         const std::set<std::string> & aliases = {})
```

Comment:
Is this actually used? It looks like aggregate initialization is used everywhere, so there’s probably no need to change the constructor

Reply:
What would you suggest instead?
Put a default value (`true`) and override that in the `Setting` constructor where needed?

I feel like modifying the constructor is better, and less error-prone (no ordering problems)

Reply:
Nah, I just have the impression that the constructor is unused − `BaseSetting { x, y , z }` doesn’t use the constructor, it just initializes the fields as they appear in the struct. But it doesn’t really matter anyways